Source/WebCore/loader/ResourceMonitor.cpp

```diff
@@ -128,7 +128,12 @@ void ResourceMonitor::checkNetworkUsageExcessIfNecessary()
 
     if (m_networkUsage.hasOverflowed() || m_networkUsage >= m_networkUsageThreshold) {
         m_networkUsageExceed = true;
-        protectedFrame()->networkUsageDidExceedThreshold();
+
+        // If the frame has sticky user activation, don't do offloading.
+        if (RefPtr protectedWindow = protectedFrame()->window(); protectedWindow && protectedWindow->hasStickyActivation())
```

Comment:
Please store the frame in a local variable to avoid repeated calls to `protectedFrame()`